### ISSUE TITLE
App is not working when using with react-native-reanimated > 2.0 on iOS

### ISSUE THREAD

**Issue by supershik** [bortolilucas/react-native-jitsimeet#4]
Hi everyone.
I tried to integrate this library in the existing project. the old project is using react-navigation
Strangly, once installing react-native-jitsimeet, app is frozen and not working anymore on iOS. but working fine on Android.

This is package.json file

    "@bortolilucas/react-native-jitsimeet": "^1.3.3",
    "@react-native-async-storage/async-storage": "^1.15.9",
    "@react-native-community/audio-toolkit": "^2.0.3",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-community/netinfo": "^7.1.7",
    "@react-native-community/push-notification-ios": "^1.10.1",
    "@react-native-firebase/app": "^12.9.2",
    "@react-native-firebase/auth": "^12.9.2",
    "@react-native-firebase/firestore": "^12.9.2",
    "@react-native-firebase/messaging": "^12.9.2",
    "@react-native-firebase/storage": "^12.9.2",
    "@react-navigation/bottom-tabs": "^6.0.9",
    "@react-navigation/drawer": "^6.1.8",
    "@react-navigation/native": "^6.0.6",
    "@react-navigation/stack": "^6.0.11",
    "add": "^2.0.6",
    "axios": "^0.23.0",
    "moment": "^2.29.1",
    "react": "16.13.1",
    "react-native": "0.63.4",
    "react-native-agora": "^3.5.1",
    "react-native-app-intro-slider": "^4.0.4",
    "react-native-background-timer": "^2.4.1",
    "react-native-callkeep": "^4.3.1",
    "react-native-country-picker-modal": "^2.0.0",
    "react-native-device-info": "^8.4.1",
    "react-native-elements": "^3.4.2",
    "react-native-fast-image": "^8.5.11",
    "react-native-gesture-handler": "^1.10.3",
    "react-native-iap": "^7.5.0",
    "react-native-image-crop-picker": "^0.36.4",
    "react-native-image-progress": "^1.1.1",
    "react-native-image-resizer": "^1.4.5",
    "react-native-keep-awake": "^4.0.0",
    "react-native-keyboard-aware-scroll-view": "^0.9.4",
    "react-native-paper": "^4.9.2",
    "react-native-popup-menu": "^0.15.11",
    "react-native-push-notification": "^8.1.1",
    "react-native-reanimated": "2.2.3",
    "react-native-safe-area-context": "^3.3.2",
    "react-native-screens": "^3.8.0",
    "react-native-share": "^7.2.0",
    "react-native-size-matters": "^0.4.0",
    "react-native-spinkit": "^1.5.1",
    "react-native-uuid": "^2.0.1",
    "react-native-vector-icons": "^8.1.0",
    "react-native-webview": "^11.14.1",
    "react-redux": "^7.2.5",
    "redux": "^4.1.1",
    "redux-thunk": "^2.3.0"
    
Is there any solution to solve this?
Thanks in advance

**Comment by bortolilucas** [comment]
It seems to be a problem with JitsiMeetSDK which is using an older version of react-native-navigation.

Removing react-native-reanimated or using v1 of react-native-reanimated might solve the problem.

I'll keep it posted once I find another solution.

More information in this discussion [#332](https://github.com/skrafft/react-native-jitsi-meet/issues/312)

**Comment by supershik** [comment]
Yes. might use v1 of react-native-reanimated but if it is used, it does not match the recent @react-navigation 5. x and 6. x.
so I'd like to use a version of react-native-reanimated 2.0+

**Comment by alien9996** [comment]
I have the same problem. And it can't even run on android. Is there any quick fix as I really need it. Help.

**Comment by bortolilucas** [comment]
Sorry guys. Since the problem is with JitsiMeetSDK, there isn't much I can do until they solve this problem. What worked for me is dropping to react-native-reanimated v1

**Comment by bortolilucas** [comment]
react-navigation still has support to react-navigation v1 on their latest versions

**Comment by alien9996** [comment]
ohh. Thank @bortolilucas . Glad you created this library, hope it will work really well in the future.

**Comment by aaryavrate** [comment]
Any solution of using reanimated@2 with this lib?

**Comment by jplandry908** [comment]
I am on v2.1.2 of this module, and using these two modules (and a lot others) in my application.

```
"react-native-reanimated": "^2.7.0",
"react-native-gesture-handler": "^2.3.2",
```

iOS is working perfectly fine without any issues, but I cannot get Android to work.

I need to exclude all of the following packages that I am using within my application or I receive duplicate errors, so I am defining the dependency like this:

```
    implementation(project(':bortolilucas_react-native-jitsimeet')) {
        exclude group: 'com.facebook.react',module:'react-native-async-storage_async-storage'
        exclude group: 'com.facebook.react',module:'react-native-async-storage'
        exclude group: 'com.facebook.react',module:'react-native-vector-icons'
        exclude group: 'com.facebook.react',module:'react-native-device-info'
        exclude group: 'com.facebook.react',module:'react-native-pager-view'
        exclude group: 'com.facebook.react',module:'react-native-reanimated'
        exclude group: 'com.facebook.react',module:'react-native-gesture-handler'
        exclude group: 'com.facebook.react',module:'react-native-screens'
        exclude group: 'com.facebook.react',module:'react-native-safe-area-context'
        exclude group: 'com.facebook.react',module:'react-native-webview'
    }
```

However, Android is crashing when I try to launch Jitsi with the following error:

```
java.lang.NoClassDefFoundError: Failed resolution of: Lcom/swmansion/gesturehandler/react/RNGestureHandlerPackage;
	at org.jitsi.meet.sdk.ReactInstanceManagerHolder.getReactNativePackages(ReactInstanceManagerHolder.java:105) ...
	
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.swmansion.gesturehandler.react.RNGestureHandlerPackage" on path: DexPathList ...
```

If anyone has any advice, I would greatly appreciated.

@bortolilucas  - Have you tried iOS since you upgraded your module to the Jisti v5 SDK? I didn't have any issues with iOS at all, but have been struggling to get Android to work for the past couple of days.

**Comment by bortolilucas** [comment]
Try to match your packages version with Jitsi's packages

https://github.com/jitsi/jitsi-meet/blob/master/package.json

**Comment by jplandry908** [comment]
Does this use the SDK v5.0.2?

https://github.com/jitsi/jitsi-maven-repository/blob/master/releases/org/jitsi/react/jitsi-meet-sdk/5.0.2/jitsi-meet-sdk-5.0.2.pom

It appears that the packages listed in this URL is different than the URL you posted.

**Comment by jplandry908** [comment]
@bortolilucas - Your suggestion led me to the correct versions I needed to get it to work. I first attempted to use the exact versions in the 'package.json' url you posted, but I was receiving an error on gesture-handler when trying to build. I was finally able to get it working with the following versions of these two modules.

```
react-native-gesture-handler@2.2.0
react-native-reanimated@1.13.4
```

Moving to react-native-reanimated v2 (any version) and gesture-handler v2.3+ produced errors. This was the highest I could go in both modules without any issues (in mode 1) for Android. I was able to get iOS to work with the current versions on both modules without any problems. Now I just need to test both operating systems thoroughly to ensure they are operating as expected. Thank you for your suggestion and your efforts in supporting this project.
